llmfoundry/data/packing.py

```diff
@@ -389,7 +390,17 @@ def profile_packing(
 
     # If streaming dataset, use a temporary local folder for profiling
     if dataloader_cfg.dataset.get('remote') is not None:
-        dataloader_cfg.dataset.local = tempfile.TemporaryDirectory().name
+        tmp_path_to_broadcast = tempfile.TemporaryDirectory().name
+        gathered_paths = dist.all_gather_object(tmp_path_to_broadcast)
+        tmp_path = gathered_paths[0]
+        dataloader_cfg.dataset.local = tmp_path
+
+    if dataloader_cfg.dataset.get('streams') is not None:
+        for stream_config in dataloader_cfg.dataset.streams.values():
+            tmp_path_to_broadcast = tempfile.TemporaryDirectory().name
+            gathered_paths = dist.all_gather_object(tmp_path_to_broadcast)
+            tmp_path = gathered_paths[0]
+            stream_config.local = tmp_path
```

Comment:
This will take global rank 0 path and fail on multi-node. You want global rank minus local rank indexed into gathered_paths

Reply:
wait actually this works (although not the way I intended) because streaming does not require that your local path already exist.

Reply:
multi node run that works with the current code: `multi-stream-18-Ah40Ce`.

Reply:
@knighton is so good he predicts user errors before they happen and prevents it

We should still use local path

Reply:
fixed, and test run: `multi-stream-19-EM0XH8`